html/mycakeapp/src/Controller/AuctionController.php

```diff
@@ -163,22 +170,124 @@ public function home()
 	{
 		// 自分が落札したBidinfoをページネーションで取得
 		$bidinfo = $this->paginate('Bidinfo', [
-			'conditions'=>['Bidinfo.user_id'=>$this->Auth->user('id')], 
+			'conditions' => ['Bidinfo.user_id' => $this->Auth->user('id')],
 			'contain' => ['Users', 'Biditems'],
-			'order'=>['created'=>'desc'],
-			'limit' => 10])->toArray();
+			'order' => ['created' => 'desc'],
+			'limit' => 10
+		])->toArray();
 		$this->set(compact('bidinfo'));
+		if ($this->request->is('post')) {
+			$buyer_status_id = (intval($this->request->getData(['buyer'])));
+			$entity = $this->BuyerStatus->get($buyer_status_id);
+			$this->BuyerStatus->patchEntity($entity, ['is_received' => true]);
+			$this->BuyerStatus->save($entity);
+			if ($this->BuyerStatus->save($entity)) {
+				$bidinfo_array = $this->Bidinfo->find('all')->toArray();
+				$count = count($this->Bidinfo->find('all')->toArray());
+				for ($i = 0; $i <= $count; $i++) {
+					if ($bidinfo_array[$i]['biditem_id'] === $entity["biditem_id"]) {
+						$bidinfo_id = $bidinfo_array[$i]['id']; //bidinfoのid
+						break;
+					}
+				}
+				$bidmsg = $this->Bidmessages->newEntity();
+				$bidmsgs = ['bidinfo_id' => $bidinfo_id, 'user_id' => $entity["buyer_id"], 'message' => '落札者が商品を受け取りました'];
+				$bidmsg = $this->Bidmessages->patchEntity($bidmsg, $bidmsgs);
+				// Bidmessageに発送完了を保存
+				$this->Bidmessages->save($bidmsg);
+				return $this->redirect(['action' => 'msg', $bidinfo_array[$i]['id']]);
+			}
+		}
+		$buyer_status = $this->BuyerStatus->find('all')->toArray();
+		$ratings = $this->Ratings->find('all')->toArray();
+		$this->set(compact('buyer_status', 'ratings'));
 	}
 
 	// 出品情報の表示
 	public function home2()
 	{
 		// 自分が出品したBiditemをページネーションで取得
 		$biditems = $this->paginate('Biditems', [
-			'conditions'=>['Biditems.user_id'=>$this->Auth->user('id')], 
+			'conditions' => ['Biditems.user_id' => $this->Auth->user('id')],
 			'contain' => ['Users', 'Bidinfo'],
-			'order'=>['created'=>'desc'],
-			'limit' => 10])->toArray();
+			'order' => ['created' => 'desc'],
+			'limit' => 10
+		])->toArray();
 		$this->set(compact('biditems'));
+
+		if ($this->request->is('post')) {
+			$biditems_id = (intval($this->request->getData(['item'])));
+			$entity = $this->Biditems->get($biditems_id);
+			$this->Biditems->patchEntity($entity, ['is_sent' => true]);
+			$this->Biditems->save($entity);
+			if ($this->Biditems->save($entity)) {
+				$bidinfo_array = $this->Bidinfo->find('all')->toArray();
+				$count = count($this->Bidinfo->find('all')->toArray());
```

Comment:
変数を使ってDB接続処理を減らしましょう
```suggestion
				$count = count($bidinfo_array);
```